generic_config_updater/patch_sorter.py

```diff
@@ -0,0 +1,965 @@
+
+        while moves:
+            move = moves.popleft()
+            if move in processed_moves:
+                continue
+            processed_moves.add(move)
+            yield move
+            moves.extend(self._extend_moves(move, diff))
+
+    def validate(self, move, diff):
+        for validator in self.move_validators:
+            if not validator.validate(move, diff):
+                return False
+        return True
+
+    def simulate(self, move, diff):
+        return diff.apply_move(move)
+
+    def _generate_moves(self, diff):
+        for generator in self.move_generators:
+            for move in generator.generate(diff):
+                yield move
+
+    def _extend_moves(self, move, diff):
+        for extender in self.move_extenders:
+            for newmove in extender.extend(move, diff):
+                yield newmove
+
+class DeleteWholeConfigMoveValidator:
+    def validate(self, move, diff):
+        if move.op_type == OperationType.REMOVE and move.path == "":
+            return False
+        return True
+
+class FullConfigMoveValidator:
+    def __init__(self, config_wrapper):
+        self.config_wrapper=config_wrapper
+
+    def validate(self, move, diff):
+        simulated_config = move.apply(diff.current_config)
+        return self.config_wrapper.validate_config_db_config(simulated_config)
+
+# TODO: Add this validation YANG models instead
+class UniqueLanesMoveValidator:
+    def validate(self, move, diff):
+        simulated_config = move.apply(diff.current_config)
+
+        if "PORT" not in simulated_config:
+            return True
+
+        ports = simulated_config["PORT"]
+        existing = set()
+        for port in ports:
+            attrs = ports[port]
+            if "lanes" in attrs:
+                lanes_str = attrs["lanes"]
+                lanes = lanes_str.split(", ")
+                for lane in lanes:
+                    if lane in existing:
+                        return False
+                    existing.add(lane)
+        return True
+
+# TODO: Add 'create-only' extension to YANG models
+class CreateOnlyMoveValidator:
+    def __init__(self, path_addressing):
+        self.path_addressing=path_addressing
+
+    def validate(self, move, diff):
+        if move.op_type != OperationType.REPLACE:
+            return True
+
+        # The 'create-only' field needs to be common between current and simulated anyway but different.
+        # This means it is enough to just get the paths from current_config, paths that are not common can be ignored.
+        paths = self._get_create_only_paths(diff.current_config)
+        simulated_config = move.apply(diff.current_config)
+
+        for path in paths:
+            tokens = self.path_addressing.get_path_tokens(path)
+            if self._value_exist_but_different(tokens, diff.current_config, simulated_config):
+                return False
+
+        return True
+
+    def _get_create_only_paths(self, config):
```

Comment:
add a comment mentioning that:
`TODO: create-only fields are hard-coded for now, it should be moved to YANG models`

remove todo at the beginning of the class